File: Two Pointers/E-283. Move Zeroes/E-283. Move Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """

        l, r = 0, 0
        zeros = []
        while r < len(nums):
            while r < len(nums) and nums[r] == 0:
                zeros.append(0)
                r += 1

            if r < len(nums):
                nums[l] = nums[r]
                l += 1
                r += 1
        nums[len(nums) - len(zeros) : len(nums)] = [*zeros]

File: Two Pointers/E-283. Move Zeroes/E-283. Move Zeroes.ts
/**
 Do not return anything, modify nums in-place instead.
 */
function moveZeroes(nums: number[]): number[] {
  const n = nums.length;
  let [l, r] = [0, 0];
  let zeros = 0;

  while (r < n) {
    while (nums[r] === 0) {
      r++;
      zeros++;
    }
    nums[l++] = nums[r++];
  }

  nums.splice(n - zeros, zeros, ...Array(zeros).fill(0));
  return nums;
}

console.log(moveZeroes([0]));
